library/core/Cargo.toml

```diff
@@ -25,6 +25,8 @@ optimize_for_size = []
 debug_refcell = []
 # Make `TypeId` store a reference to the name of the type, so that it can print that name.
 debug_typeid = []
+#Allow -Zc-char-type='unsigned|signed|default' for c_char override
```

Comment:
Why do we need a compiler flag? As far as I can see, this (and the other change in `library/core/src/ffi/primitives.rs`) are the only things actually needed. Given libcore is precompiled, I don't understand what effect `-Zc-char-type` could actually have.

Reply:
This option also need to be consistent with libc c_char. With flag, this consistency will be provided by target modifiers system.
May feature flag mechanics secure consistency of core c_char with libc c_char?
Is there an example of existing feature flag with similar logic?

Reply:
The Rust compiler doesn't have special knowledge of the `libc` crate so it's not really in a position to enforce that. Maybe libc should just re-export the type from `core::ffi`? 

cc @tgross35 since you've been working on the libc crate lately 🙂

Reply:
I'd like to reexport `core::ffi` for now and then just drop the libc types entirely for 1.0, but we can't make these changes yet due to MSRV and compat concerns. That should eventually be able to happen though so I wouldn't make any decisions here based on `libc`, IOW a library flag seems reasonable 👍 (assuming the compiler doesn't need knowledge of this for whatever reason).

Some more details: https://github.com/rust-lang/libc/issues/4257